### ISSUE TITLE
Library has issues on the esp32 platform.

### ISSUE THREAD

**Issue by bperrybap** [Nospampls/SchedTask#4]
There are some issues in the library that prevent it from compiling on the esp32 platform.
If you install the esp32 platform and compile Example_1 for the board NodeMCU-32s with warnings enabled you will see the issues / errors.

One thing that is causing some of these issues, is that compilation flags being used on the esp32 platform are stricter than some of the other platforms.
 i.e. issues that might be treated as warnings on other platforms are treated as errors on this platform.
(which is a good thing in IMO, as I believe many of the other platforms are not strict enough)
Some of the issues are actual real bugs like the not returning anything from a function that is supposed to return something. In this case the function would return garbage to the caller.

You can see the issues/errors when you do the build with warnings enabled, but here are a few I've seen:
- **addTask() must return something**
We touched on this in issue #1 for the esp8266 platform

- **SoftwareSerial.h should not be included**
It does not appear to be needed by the library code or examples.
Since not all platforms have SoftwareSerial (like the esp32), including it can cause the library to not compile.

- **nested comments**
There are some issues with use of /* */ in the examples.
These are considered errors on the esp32 platform.
I prefer using either // on every line
or to "mark" comment section like this:
```
/******************************************************
 * Here is a "marked" comment section
 * and another line
 * and another line
 ******************************************************/
```
or use a multi line comment like this:
```
/*
 * a multi line comment
 * another line of the multi line comment
 */
```

- **issues with virtual function declarations and default definitions**
I'm not sure what you were intending with your virtual function declarations  and definitions.
They all have an empty default function and then an extra semicolon after the ending curly brace.
And in particular the function getFunc() has an error in its default definition in that it does not return anything.
If the virtual functions must exist then there is no need for a default definition and you could declare them like this:

`		virtual void setFunc(pFunc);`
`		virtual pFunc getFunc();`
`		virtual void callFunc();`

If you need a default class function that can be overridden, then create the default function using curly braces but leave off the semicolon after the right curly brace like this:

`		virtual void setFunc(pFunc){}`
`		virtual pFunc getFunc(){return((pFunc)0);}`
`		virtual void callFunc(){}`

**Comment by Nospampls** [comment]
- addTask -- already addressed/fixed
   - SoftwareSerial -- I was unaware that I did not need it for Serial <<
   "blah"; so I removed all #includes.
   - nested comments -- I have not discovered where this is an issue but
   I'll look some more.  Easily fixed.
   - virtual functions -- I do not need a default definition so will try to
   use your first suggestions.

However, I have an issue with virtual functions getFunc() and setFunc()
because the function for SchedTaskT takes a parameter and the function for
SchedTask does not.  The result is that the signatures are different.

What I want is if the user has a pointer to a scheduled task for him to use
getFunc() regardless of whether the object is SchedTask or SchedTaskT.

SchedTask Task1 (NOW, ONESHOT, task1);
SchedTaskT<int> Task2 (NOW, ONESHOT, task2, 1);
....
SchedBase* pTask;
pTask = &Task1;

void (*pF)() = *pTask->getFunc()*;

pTask = &Task2;
void (*pF2)(int) = *pTask->getFunc()*;

*...*
void task1 () {;}
void task2 (int parm) {;}


The above compiles (provided I incorrectly use the pFunc instead of the
pFuncT type in the SchedTaskT class) with warnings for Uno/Mega but errors
for NodeMCU-32S.

Any suggestions would be appreciated.  I'd like to find a solution that
does not break API or documentation.

Thanks.
--rolfe.
